lib/claim/token-holder-revenue-fund-contracts-ensemble.js

```diff
@@ -0,0 +1,359 @@
+'use strict';
+
+const {utils: {bigNumberify}} = require('ethers');
+const TokenHolderRevenueFundContract = require('./token-holder-revenue-fund-contract');
+const NestedError = require('../nested-error');
+
+const _ensemble = new WeakMap();
+const _spannedByCurrency = new WeakMap();
+const _firstAccrualOffset = new WeakMap();
+
+const currencyToKey = (currency) => `${currency.ct.toString()}-${currency.id}`;
+
+const minBigNumber = function (n1, n2) {
+    return n1.lt(n2) ? n1 : n2;
+};
+
+/**
+ * @class TokenHolderRevenueFundContractsEnsemble
+ * An ensemble of instances of TokenHolderRevenueFundContract
+ * @alias module:nahmii-sdk
+ */
+class TokenHolderRevenueFundContractsEnsemble {
+    constructor(walletOrProvider, abstractionNames = ['TokenHolderRevenueFund'], firstAccrualOffset = 0) {
+        _ensemble.set(this, abstractionNames.map(n => ({contract: new TokenHolderRevenueFundContract(walletOrProvider, n)})));
+        _spannedByCurrency.set(this, new Map());
+        _firstAccrualOffset.set(this, firstAccrualOffset > 0 ? firstAccrualOffset : 0);
+    }
+
+    /**
+     * Returns the first accrual's offset index used in the (temporal) span
+     * @returns {number} - First accrual's offset
+     */
+    get firstAccrualOffset() {
+        return _firstAccrualOffset.get(this);
+    }
+
+    /**
+     * Return status of whether contract instance span has been executed
+     * for the given currency
+     * @param {Currency} currency - The currency
+     * @returns {boolean} True if span has been executed, else false
+     */
+    isSpanned(currency) {
```

Comment:
Should this function be public?

Reply:
I am removing from the public interface. 👍